play.pokemonshowdown.com/src/client-connection.ts

```diff
@@ -19,6 +19,7 @@ export class PSConnection {
 	private shouldReconnect = true;
 	reconnectTimer: ReturnType<typeof setTimeout> | null = null;
 	private worker: Worker | null = null;
+	battleRoomsToRejoin: string[] = [];
```

Comment:
I don't think this should be state. You can just iterate all rooms when you need this list.